test/extension-tests/successful-build/test/configure-and-build.test.ts

```diff
@@ -83,7 +83,7 @@ suite('Build', async () => {
   });
 
   test('Configure', async () => {
-    expect(await cmt.configure()).to.be.eq(0);
+    expect(await cmt.configure([], ConfigureType.Normal, ConfigureTrigger.runTests)).to.be.eq(0);
```

Comment:
I probably would have made the trigger the first parameter since all are optional. This would also make it match `driver.configure`

Reply:
Making the trigger the first parameter causes compilation errors with the configure prototype in CMake Tools API. I changed the prototype to be able to achieve what you suggest. Luckily not a breaking change since the first parameter I added is still optional. We'll discuss if there is a better way.

Reply:
The CMake Tools API is not currently enabled for external callers and has some problems.  There is a "leak" in the API.  `type` is not part of the declaration, but it is in the definition.

What I think needs to happen here is we need to forward the public API call to an internal version of `configure` that accepts additional parameters.  When calling the public version it will automatically add the correct `trigger` and `type`.